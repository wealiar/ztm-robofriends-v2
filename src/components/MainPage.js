import React, { Component } from 'react';
import CardList from './CardList';
import Header from './Header';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import './MainPage.css'

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    }

    render() {
        const { onSearchChange, isPending } = this.props;

        return (isPending)
            ? <h1 className='tc'> Loading </h1>
            : (
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundry>
                            <CardList robots={this.filterRobots()} />
                        </ErrorBoundry>
                    </Scroll>
                </div>
            );
    }
}

export default MainPage;