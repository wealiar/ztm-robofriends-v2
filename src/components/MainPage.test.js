import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false,
    }
    wrapper = shallow(<MainPage {...mockProps} />);
});


describe('MainPage smart component', () => {
    it('expect to render MainPage component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('filters robots correctly', () => {
        expect(wrapper.instance().filterRobots()).toEqual([]);

        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com',
            }],
            searchField: 'john',
            isPending: false,
        };

        const wrapper2 = shallow(<MainPage {...mockProps2} />);
        expect(wrapper2.instance().filterRobots()).toEqual([{
            id: 3,
            name: 'John',
            email: 'john@gmail.com',
        }]);
    });

    it('filters robots correctly 2', () => {
        expect(wrapper.instance().filterRobots()).toEqual([]);

        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com',
            }],
            searchField: 'a',
            isPending: false,
        };

        const filteredRobots = [];

        const wrapper3 = shallow(<MainPage {...mockProps3} />);
        expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
    });

    it('getting loading screen when pending', () => {
        const mockProps3 = {
            onRequestRobots: jest.fn(),
            isPending: true,
        };

        const wrapper3 = shallow(<MainPage {...mockProps3} />);
        expect(wrapper3.instance().render()).toEqual(<h1 className='tc'> Loading </h1>);
    });
    
});
