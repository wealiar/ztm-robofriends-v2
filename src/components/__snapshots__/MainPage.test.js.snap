// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage smart component expect to render MainPage component 1`] = `ShallowWrapper {}`;
